sympy/stats/crv.py

```diff
@@ -346,6 +346,10 @@ def probability(self, condition, **kwargs):
             # return S.Zero if `domain` is empty set
             if domain.set is S.EmptySet or isinstance(domain.set, FiniteSet):
                 return S.Zero if not cond_inv else S.One
+            if isinstance(domain.set, Union):
+                return sum(
+                     Integral(pdf(z), (z, subset), **kwargs) for subset in
+                     domain.set.args if isinstance(subset, Interval))
```

Comment:
What role does `isinstance(subset, Interval)` have here? Anything that's not an Interval is simply ignored. Is this the right thing to do?

Reply:
Please correct me if I am wrong, but with the current implementation of `where`,  if `domain.set` is a union, each instance would be a `FiniteSet` or an `Interval`. Being continuous RVs, only the intervals contribute to the probability of the event. If `subset` is a `FiniteSet`, an error would be raised because of incorrect integration limits.
Should I do something like summing up the probability over each `subset`, and if an error is raised while caculating the integral, just add the unevaluated object to the total probability?

Reply:
Thanks for explanation. Indeed, I can't think of a situation where `domain.set` would be a set of positive measure that's not a union of intervals. Things like a Cantor set of positive measure can hardly arise from `where`, and even if they did, they would not be representable by a Union.